source/quic_server_ssl_poll_http.c

```diff
@@ -0,0 +1,3286 @@
+ * defines the size payload sent by server.
+ */
+#include <string.h>
+#include <stdio.h>
+#include <assert.h>
+#include <ctype.h>
+#include <sys/types.h>
+#include <time.h>
+
+/* Include the appropriate header file for SOCK_STREAM */
+#ifdef _WIN32 /* Windows */
+# include <winsock2.h>
+# include "perflib/basename.h"
+# include "perflib/err.h"
+#else /* Linux/Unix */
+# include <err.h>
+# include <sys/socket.h>
+# include <sys/select.h>
+# include <netinet/in.h>
+# include <unistd.h>
+# include <poll.h>
+# include <libgen.h>
+#endif
+
+#include <openssl/bio.h>
+#include <openssl/ssl.h>
+#include <openssl/err.h>
+#include <openssl/quic.h>
+#include "perflib/perflib.h"
+
+#ifndef _WIN32
+# include <unistd.h>
+#else
+# include <windows.h>
+# include "perflib/getopt.h"
+#endif /* _WIN32 */
+
+/*
+ * The code here is based on QUIC poll server found in demos/quic/poll-server
+ * in OpenSSL source code repository. Here we take the demo one step further
+ * and implement also non-blocking client which talks to server to measure
+ * performance.
+ *
+ * Server accepts QUIC connections. It then accepts bi-directional
+ * stream from client and reads request. By default it sends
+ * 12345 bytes back as HTTP/1.0 response to any GET request.
+ * If GET request comes with URL for example as follows:
+ *     /foo/bar/file_65535.txt
+ * then the server sends 64kB of data in HTTP/1.0 response.
+ */
+
+#define DEBUG 1
+#ifdef DEBUG
+# define DPRINTF fprintf
+# define DPRINTFC fprintf
+# define DPRINTFS fprintf
+#else
+# define DPRINTF(...) (void)(0)
+# define DPRINTFC(...) (void)(0)
+# define DPRINTFS(...) (void)(0)
+#endif
+
+/*
+ * format string so we can print SSL_poll() events in more informative
+ * way. To print events one does this:
+ *   int events = SSL_POLL_EVENT_F | SSL_POLL_EVENT_R;
+ *   printf("%s We got events: " POLL_FMT "\n", __func__, POLL_PRINTA(events));
+ */
+#define POLL_FMT "%s%s%s%s%s%s%s%s%s%s%s%s%s"
+#define POLL_PRINTA(_revents_) \
+    (_revents_) & SSL_POLL_EVENT_F ? "SSL_POLL_EVENT_F " : "", \
+    (_revents_) & SSL_POLL_EVENT_EL ? "SSL_POLL_EVENT_EL " : "", \
+    (_revents_) & SSL_POLL_EVENT_EC ? "SSL_POLL_EVENT_EC " : "", \
+    (_revents_) & SSL_POLL_EVENT_ECD ? "SSL_POLL_EVENT_ECD " : "", \
+    (_revents_) & SSL_POLL_EVENT_ER ? "SSL_POLL_EVENT_ER " : "", \
+    (_revents_) & SSL_POLL_EVENT_EW ? "SSL_POLL_EVENT_EW " : "", \
+    (_revents_) & SSL_POLL_EVENT_R ? "SSL_POLL_EVENT_R " : "", \
+    (_revents_) & SSL_POLL_EVENT_W ? "SSL_POLL_EVENT_W " : "", \
+    (_revents_) & SSL_POLL_EVENT_IC ? "SSL_POLL_EVENT_IC " : "", \
+    (_revents_) & SSL_POLL_EVENT_ISB ? "SSL_POLL_EVENT_ISB " : "", \
+    (_revents_) & SSL_POLL_EVENT_ISU ? "SSL_POLL_EVENT_ISU " : "", \
+    (_revents_) & SSL_POLL_EVENT_OSB ? "SSL_POLL_EVENT_OSB " : "", \
+    (_revents_) & SSL_POLL_EVENT_OSU ? "SSL_POLL_EVENT_OSU " : ""
+
+/*
+ * Code below comes from sys/queue.h found on OpenBSD. The head member
```

Comment:
Code below how far? Please indicate in the code comments where this ends. It might be cleaner to pull this out into a separate header file to avoid any confusion. I don't really like having different licence terms in the middle of a file.

Are we ok with including this here? We could not do this in the main repo without CLAs from all authors.